### PR TITLE
Init command

### PR DESCRIPTION
**Pull request by BradfordMedeiros**
Made new pr.  rebasing had was messy, easier to just open new pr. 
Added templates instead of the simple string replace.  

Still working on the extra patching and test suite integration on this pr and that can either be in this or a separate pr, doesn't matter from my perspective.

### CODE REVIEW

File: pkg/commands/generate/get_command.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+func GetCommand() *cobra.Command {
+	var filepath string
+	var componentName string
+	cmd := &cobra.Command{
+		Use:   "generate",
+		Short: "Generate placeholder components",
+		Long:  "Generate placeholder components",
+		Run: func(cmd *cobra.Command, _ []string) {
+			if filepath == "" {
+				os.Stderr.WriteString("Filepath not specified")
```
Comment (Kashomon) [inline pkg/commands/generate/get_command.go]: log.Exit("....")

File: pkg/commands/generate/get_command.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+				os.Exit(1)
+			}
+			generate.Create(filepath, componentName)
+		},
+	}
+	cmd.Flags().StringVar(&filepath, "path", "", "filepath to create component")
```
Comment (Kashomon) [inline pkg/commands/generate/get_command.go]: how about "write-to" to make this consistent with the `export` command? (well, when that lands).

I'd also mention that this is required

File: pkg/commands/generate/get_command.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+package generate
+
+import (
+	"github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/generate"
+	"github.com/spf13/cobra"
+	"os"
+)
+
+// GetCommand generate placeholder components
+func GetCommand() *cobra.Command {
```
Comment (Kashomon) [inline pkg/commands/generate/get_command.go]: This should receive dependencies (file-writer).

File: pkg/generate/generate.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+      type: string
+      pattern: '^[a-z0-9-]+$'
+`
+const patchOptions = `# Options for applying to the component via patch-templates
+namespace: foo-namespace
+buildLabel: test-build
+`
+
+// Create scaffolds basic set of files to the filesystem
+func Create(filepath string, name string) error {
```
Comment (Kashomon) [inline pkg/generate/generate.go]: I'd just return map[string]contents and write out the data in the Command. Limits the I/O to the commands and makes this function easier to test.

Alternatively, pass in the filereaderwriter

File: pkg/generate/generate.go [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+		goto handleError
+	}
+	if err := ioutil.WriteFile(path.Join(filepath, "sample-options.yaml"), []byte(patchOptions), 0666); err != nil {
+		writeErr = err
+		goto handleError
+	}
+
+handleError:
+	if writeErr != nil {
+		os.Remove(filepath)
```
Comment (Kashomon) [inline pkg/generate/generate.go]: I think it's fine to write partial results. Also, the go-to stuff is pretty weird

File: pkg/commands/generate/get_command.go (hunk 3 above)
Comment (BradfordMedeiros) [reply pkg/commands/generate/get_command.go]: Not going to use file writer in this code unless we remove references to bundle in filewriter. 
No reason for this code to depend on the bundle implementation.

File: pkg/generate/generate.go (hunk 5 above)
Comment (BradfordMedeiros) [reply pkg/generate/generate.go]: okay.  I think in practice that should work well.

**Comment by Kashomon** [comment]
/lgtm 
This is a good start. we can iterate on this.

**Pull request merged**